drivers/block-crypto.c

```diff
@@ -143,7 +143,7 @@ find_keyfile(char **keyfile, const char *dirs,
 			safe_strncpy(keydir, dirs, sizeof(keydir));
 			dirs = NULL;
 		} else {
-			size_t len = sep - dirs;
+			size_t len = (sep - dirs) + 1;
```

Comment:
This may fix the newly introduced bug but the code is still buggy because nothing guarantees that `len < sizeof(keydir)`. I think something like this should work:

```
size_t len = sep - dirs;
strncpy(keydir, dirs, MIN(len, sizeof(keydir) - 1));
...
```

This should result in no more than 255 characters copied and the 256th char is guaranteed to be NUL since keydir is zero-initialized.

Reply:
I can change it to add that guarantee.  Also just to clarify, I assume you want to keep the call to `safe_strncpy` instead of back to the old `strncpy`?

Reply:
@crogers1 tbh given that this code is exclusively within an `#ifdef OPEN_XT` we really couldn't care less about whether you use `strncpy` or `safe_strncpy` as we don't even compile this code.

Reply:
Understood. I'd still like to keep it consistent with the rest of blktap so I'll leave the `safe_strncpy` in.  Appreciate y'all giving this a review.